health/notifications/alarm-notify.sh.in

```diff
@@ -2179,7 +2179,7 @@ send_opsgenie() {
 EOF
 )
 
-  httpcode=$(docurl -X POST -H "Content-Type: application/json" -d "${payload}" "https://api.opsgenie.com/v1/json/integrations/webhooks/netdata?apiKey=${OPSGENIE_API_KEY}")
+  httpcode=$(docurl -X POST -H "Content-Type: application/json" -d "${payload}" "${OPSGENIE_API_URL}/v1/json/integrations/webhooks/netdata?apiKey=${OPSGENIE_API_KEY}")
```

Comment:
This needs to be explicitly set to a default value earlier in the file (preferrably right around line 393, where we define the API key variable) so that OpsGenie support does not break for existing users who have an otherwise customized version of the config file (essentially, we cannot rely here on the config file defining this variable for us, because the user may have an old version of the config file with modifications and would have to manually update it to et this working again in that case).

Reply:
@tinyhammers

[shell parameter expansion](https://www.gnu.org/software/bash/manual/html_node/Shell-Parameter-Expansion.html) will do


> ${parameter:-word}
If parameter is unset or null, the expansion of word is substituted. Otherwise, the value of parameter is substituted.

Reply:
You mean like this?
`OPSGENIE_API_URL="${parameter:-https://api.opsgenie.com}"`

Reply:
- unset case, use default value

```cmd
0 ~ $ OPSGENIE_API_URL=""
0 ~ $ OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}
0 ~ $ echo $OPSGENIE_API_URL
https://api.opsgenie.com
```

- set case, use set value

```cmd 
0 ~ $ OPSGENIE_API_URL="https://api.eu.opsgenie.com"
0 ~ $ OPSGENIE_API_URL=${OPSGENIE_API_URL:-"https://api.opsgenie.com"}
0 ~ $ echo $OPSGENIE_API_URL
https://api.eu.opsgenie.com
```